js/kubewarden/host_capabilities/crypto/types.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Certificate encoding types
+ */
+export type CertificateEncoding = 'Der' | 'Pem';
+
+/**
+ * Certificate represents a x509 certificate
+ */
+export interface Certificate {
+  encoding: CertificateEncoding;
+  data: number[];
```

Comment:
`data` must always be utf8, we should document it on the type here.

In addition, maybe we should provide the following functions:

```js
export namespace CertificateUtils {
  export function fromString(certString: string, encoding: CertificateEncoding): Certificate {
    return {
      encoding,
      data: Array.from(new TextEncoder().encode(certString)), // always UTF-8
    };
  }

  export function toString(cert: Certificate): string {
    return new TextDecoder('utf-8').decode(Uint8Array.from(cert.data));
  }
}
```